daily_milestone.go

```diff
@@ -1,29 +1,221 @@
+var logger *log.Logger
+
+// LoggerSetup Initialization
+func LoggerSetup(info io.Writer) {
+	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+// Function to get project ID from the gitLabAPI
+func getProjectID(baseURL string, token string, projectname string, namespace string) (string, error) {
+	projects := []gitLabAPI{}
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", baseURL, nil)
+	if err != nil {
+		logger.Println(err)
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
 	}
 
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
+
+	}
+	json.Unmarshal(respByte, &projects)
+	defer resp.Body.Close()
+	for _, p := range projects {
+		// To check for message404 error and that project is not forund
+		if p.Name == "message" {
+			fmt.Println(p.Name)
+		}
+
+		if p.Name == projectname && p.NameSpace.Path == namespace {
+			fmt.Println(strconv.Itoa(p.ID))
+			return strconv.Itoa(p.ID), nil
+		}
+	}
+	return "", fmt.Errorf("project %s not found", projectname)
 }
 
-//func get_milestones(base_url string, token string, project_id string)[]string{}
-//func create_milestone_data(advance int) []string{}
-//func create_milestones(base_url string, token string, project_id string, milestones string) string{}
+// It is getting milestones data from the milestone API and returning list of closed milestones
+func getMilestones(baseURL string, token string, projectID string) ([]string, error) {
+	milestones := []milestoneAPI{}
+	list := []string{}
+	strurl := []string{baseURL, projectID, "/milestones"}
+	url := strings.Join(strurl, "")
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		logger.Println(err)
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
+	}
+
+	json.Unmarshal(respByte, &milestones)
+	defer resp.Body.Close()
+	for _, m := range milestones {
+		if m.State != "closed" {
+			titleTuple := []string{"Title: ", m.Title, ", DueDate", m.DueDate}
+			y := strings.Join(titleTuple, " ")
+			list = append(list, y)
+		}
+	}
+	fmt.Println(list)
+	return list, nil
+}
+
+// CreateMilestoneData is used to check the due date using the time package of python
+func createMilestoneData(advance int) []string {
+	today := time.Now().Local()
+	list := []string{}
+	for i := 0; i < advance; i++ {
+		date := today.AddDate(0, 0, i)                                                                      // To Format to ISOFormat and it converts to string so can be used in list directly
+		datelist := []string{"Title: ", date.Format("2006-01-02"), "  due_date", date.Format("2006-01-02")} // Was unable to get a map in a list so made this
+		y := strings.Join(datelist, ",")
```

Comment:
Same here probably, no big deal but the `y` feels like a name that came out of nowhere (unless I'm missing something